components/brave_origin/common/mojom/brave_origin_settings.mojom

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module brave_origin.mojom;
+
+// Interface for communicating with BraveOriginService from WebUI
+interface BraveOriginSettingsHandler {
+  // Check if the current user is a Brave Origin user
+  IsBraveOriginUser() => (bool is_brave_origin_user);
+
+  // Check if a specific preference is controlled by Brave Origin
+  IsPrefControlledByBraveOrigin(string pref_name) =>
+      (bool is_controlled);
+
+  // Get the current value of a Brave Origin controlled preference
+  GetBrowserPolicyValue(string pref_name) => (bool? value);
+  GetProfilePolicyValue(string pref_name) => (bool? value);
```

Comment:
It's not clear to me how the UI would know whether a policy is for the browser or profile. It also seems odd that `IsPrefControlledByBraveOrigin` works for both browser and profile policies, while these functions are separate. IMO, it would be nice if we combined these functions (so we have `GetPolicyValue` and `SetPolicyValue`) so we could simplify the UI logic. Thoughts?

Reply:
Fixed in https://github.com/brave/brave-core/pull/31404/commits/60ad806c3fd511222c385a40faf895b2847291b2